modules/fedimint-lnv2-client/src/lib.rs

```diff
@@ -375,6 +375,34 @@ impl LightningClientModule {
         Ok(operation_id)
     }
 
+    async fn get_operation_id(&self, invoice: &Bolt11Invoice) -> anyhow::Result<OperationId> {
```

Comment:
Something I totally missed but you reminded me of the other day: you probably want to do all that in a single database transaction (together with spawning the payment state machines) so that concurrent calls to the pay function don't cause havoc (can easily happen due to UI bugs that such a function is called concurrently).

Reply:
If we get a collision of operation ids do to concurrent calls with the same invoice finalize_and_submit_transaction will return the appropriate error which we return so its safe. If we do it in one dbtx we would get the more appropriate error of "Previous payment attempt still in progress.", but achieving this is seems much more involved.